docs/en/tutorials/optim_wrapper.md

```diff
@@ -1,3 +1,500 @@
+```
+
+### Use multiple optimizers
+
+Considering that algorithms like GANs usually need to use multiple optimizers to train the generator and the discriminator, OptimWrapper provides a container class called `OptimWrapperDict` to manage. `OptimWrapperDict` stores the wrappers in the form of dicts and allows users to access and traverse the elements just like a dict.
+
+Unlike regular OptimWrapper, `OptimWrapperDict` does not provide methods such as `update_prarms`, `optim_context`, `backward`, `step`, etc. Plus, it cannot be used directly for training models. We suggest to access the elements in `OptimWrapperDict` directly to implement the parameter update logic.
+
+Users may wonder why not just use `dict` to manage multiple optimizers if `OptimWrapperDict` does not have training capabilities. Actually, the core function of `OptimWrapperDict` is to support batch exporting and loading of state dicts of all OptimWrappers, and to support getting learning rates and momentums as well. Without `OptimWrapperDict`, MMEngine needs to do a lot `if-else` in OptimWrapper to get all the states of the OptimWrappers.
+
+```python
+from torch.optim import SGD
+import torch.nn as nn
+
+from mmengine.optim import OptimWrapper, OptimWrapperDict
+
+gen = nn.Linear(1, 1)
+disc = nn.Linear(1, 1)
+optimizer_gen = SGD(gen.parameters(), lr=0.01)
+optimizer_disc = SGD(disc.parameters(), lr=0.01)
+
+optim_wapper_gen = OptimWrapper(optimizer=optimizer_gen)
+optim_wapper_disc = OptimWrapper(optimizer=optimizer_disc)
+optim_dict = OptimWrapperDict(gen=optim_wapper_gen, disc=optim_wapper_disc)
+
+print(optim_dict.get_lr())  # {'gen.lr': [0.01], 'disc.lr': [0.01]}
+print(optim_dict.get_momentum())  # {'gen.momentum': [0], 'disc.momentum': [0]}
+```
+
+```
+{'gen.lr': [0.01], 'disc.lr': [0.01]}
+{'gen.momentum': [0], 'disc.momentum': [0]}
+```
+
+As shown in the above example, `OptimWrapperDict` exports learning rates and momentums for all OptimWrappers easily, and OptimWrappers can export and load all the state dicts in a similar way.
+
+### Configure the OptimWapper in [Runner](runner.md)
+
+We first need to configure the `optimizer` for the OptimWrapper. MMEngine automatically adds all optimizers in PyTorch to the `OPTIMIZERS` registry, and users can specify the optimizers they need in the form of a dict. All supported optimizers in PyTorch are listed [here](https://pytorch.org/docs/stable/optim.html#algorithms).
+
+Now we take setting up a SGD OptimWrapper as an example.
+
+```python
+optimizer = dict(type='SGD', lr=0.01, momentum=0.9, weight_decay=0.0001)
+optim_wrapper = dict(type='OptimWrapper', optimizer=optimizer)
+```
+
+Here we configured a OptimWrapper with the type of SGD, and other parameters such as learning rate, momentum, etc. Considerting that OptimWrapper is standard single-precision training, we can skip the `type` as follows.
```

Comment:
```suggestion
Here we have set up an OptimWrapper with a SGD optimizer with the learning rate and momentum parameters as specified. Since OptimWrapper is designed for standard single precision training, we can also omit the `type` field in the configuration:
```